tensorflow_probability/python/distributions/empirical_test.py

```diff
@@ -0,0 +1,694 @@
+      dist = empirical.Empirical(samples=input_ph)
+      expected_shape = tf.concat(
+          [dist.batch_shape_tensor(), dist.event_shape_tensor()],
+          axis=0)
+      self.assertAllEqual(self.evaluate(tf.shape(dist.sample())),
+                          expected_shape)
+
+      n = 1000
+      seed = 42
+      self.assertAllEqual(self.evaluate(dist.sample(n, seed)),
+                          self.evaluate(dist.sample(n, seed)))
+
+  def testMean(self):
+    sample = [[[0, 0], [0, 1]],
+              [[0, 1], [1, 2]],
+              [[0, 1], [1, 2]],
+              [[0, 2], [2, 4]]]
+
+    expected_mean = [[0, 1], [1, 2.25]]
+
+    input_ = tf.convert_to_tensor(sample, dtype=np.float32)
+    input_ph = tf.placeholder_with_default(
+        input=input_,
+        shape=input_.shape if self.static_shape else None)
+    dist = empirical.Empirical(samples=input_ph, event_ndims=2)
+    self.assertAllClose(self.evaluate(dist.mean()), expected_mean)
+
+  def testMode(self):
+    sample = [[[0, 0], [0, 1]],
+              [[0, 1], [1, 2]],
+              [[0, 1], [1, 2]],
+              [[0, 2], [2, 4]]]
+
+    expected_mode = [[0, 1], [1, 2]]
+
+    input_ = tf.convert_to_tensor(sample, dtype=np.int32)
+    input_ph = tf.placeholder_with_default(
+        input=input_,
+        shape=input_.shape if self.static_shape else None)
+    dist = empirical.Empirical(samples=input_ph, event_ndims=2)
+    self.assertAllClose(self.evaluate(dist.mode()), expected_mode)
+
+  def testEntropy(self):
+    sample = [[[0, 0], [0, 1]],
+              [[0, 1], [1, 2]],
+              [[0, 1], [1, 2]],
+              [[0, 2], [2, 4]]]
+
+    expected_entropy = entropy([0.25, 0.5, 0.25])
+
+    input_ = tf.convert_to_tensor(sample, dtype=np.float64)
+    input_ph = tf.placeholder_with_default(
+        input=input_,
+        shape=input_.shape if self.static_shape else None)
+    dist = empirical.Empirical(samples=input_ph, event_ndims=2)
+    self.assertAllClose(self.evaluate(dist.entropy()), expected_entropy)
+
+  def testVarianceAndStd(self):
+    sample = [[[0, 1], [1, 2]],
+              [[0, 3], [3, 6]]]
+
+    expected_variance = [[0, 1], [1, 4]]
+
+    input_ = tf.convert_to_tensor(sample, dtype=np.float64)
+    input_ph = tf.placeholder_with_default(
+        input=input_,
+        shape=input_.shape if self.static_shape else None)
+    dist = empirical.Empirical(samples=input_ph, event_ndims=2)
+    self.assertAllClose(self.evaluate(dist.variance()),
+                        expected_variance)
+    self.assertAllClose(self.evaluate(dist.stddev()),
+                        np.sqrt(expected_variance))
+
+
+@tf_test_util.run_all_in_graph_and_eager_modes
+class EmpiricalScalarStaticShapeTest(EmpiricalScalarTest, tf.test.TestCase):
+  static_shape = True
+
+@tf_test_util.run_all_in_graph_and_eager_modes
+class EmpiricalScalarDynamicShapeTest(EmpiricalScalarTest, tf.test.TestCase):
+  static_shape = False
+
+@tf_test_util.run_all_in_graph_and_eager_modes
```

Comment:
are we sure this is applied to test cases in extended test classes? (you could try adding a test with `self.assertFalse(tf.executing_eagerly())`)
I would have thought the eager case I pointed out above would be failing.